src/Fixer/ClassNotation/NoUnneededFinalMethodFixer.php

```diff
@@ -150,17 +150,8 @@ private function getMethods(Tokens $tokens): \Generator
 
         $enums = [];
         $classesAreFinal = [];
-        $elements = $tokensAnalyzer->getClassyElements();
-
-        for (end($elements);; prev($elements)) {
-            $index = key($elements);
-
-            if (null === $index) {
-                break;
-            }
-
-            $element = current($elements);
 
+        foreach ($tokensAnalyzer->getClassyElements() as $index => $element) {
```

Comment:
we were going from end to start originally. no need to do array_reverse here as well ?

Reply:
No, we don't need it. We only need to iterate tokens from the end if we insert some new ones (as they "push" the indexes) and we don't insert any here.